vscode/src/utils.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Guid format such as "00000000-1111-2222-3333-444444444444"
+export function getRandomGuid(): string {
```

Comment:
I think a simple sequence number (1, 2, 3...) would work here. No need to deal with GUIDs.

The [common properties](https://github.com/Microsoft/vscode-extension-telemetry#common-properties) already include the VSCode session ID. So you don't have to worry about _global_ uniqueness, just uniqueness within a VS Code session.

The extension can be deactivated and reactivated within a VS Code session, so you probably want some sort of ExtensionActivationId to differentiate between activations. For that, I think "start time" could work. (We can - and probably should - include that in `sendTelemetryEvent` as a common property for all events).

Once you have those two in each event, then you don't need to worry about global uniqueness in any of the other properties. A monotonically increasing integer should suffice.